conda-recipe/meta.yaml

```diff
@@ -1,47 +1,55 @@
-{% set data = load_setup_py_data() %}
-{% set name = "happi" %}
+{% set package_name = "happi" %}
+{% set import_name = "happi" %}
+{% set version = load_file_regex(load_file=os.path.join(import_name, "_version.py"), regex_pattern=".*version = '(\S+)'").group(1) %}
 
 package:
-    name: {{ name }}
-    version: {{ data.get('version') }}
+  name: {{ package_name }}
+  version: {{ version }}
+
+source:
+  path: ..
 
 build:
-  noarch: python
   number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
 
-source:
-    path: ..
 
 requirements:
-    build:
-      - python >=3.6
-      - pip
-    run:
-      - python >=3.6
-      - click
-      - coloredlogs
-      - entrypoints
-      - jinja2
-      - prettytable
-      - simplejson
-    run-constrained:
-      - psdm_qs_cli >=0.3.1
-      - pymongo >=4.0.2
+  build:
+  - python >=3.6
+  - pip
+  - setuptools_scm
+  run:
+  - python >=3.6
+  - click
+  - coloredlogs
+  - entrypoints
+  - jinja2
+  - prettytable
+  - simplejson
+  run-constrained:
+  - psdm_qs_cli >=0.3.1
+  - pymongo >=4.0.2
+
+
 
 test:
-    imports:
-      - happi
-    commands:
-      - happi --help
+  commands:
+  - happi --help
+  imports:
+  - happi
+  requires:
+  - pytest
+  - ipython
```

Comment:
We could add `line_profiler` here in the `test` section instead of in the workflow configuration
Similarly, we can add `line-profiler` to `dev-requirements.txt` for pip

I think it's probably a better pattern to follow in general...

Reply:
This one is weird since `line-profiler` is being pulled from `pcdsutils`, but isn't a firm requirement there either.  (we just try to import it and return if we can't)

I do think it's probably good to have an explicit requirement here since we're advertising the line-profiler functionality, but I could also see a world where we leave it in the job recipe.

Reply:
Yeah, that's fair... 

Looking back a bit, I think the original reason was that the package name differs on conda-forge and pypi. For our Travis CI builds, we erroneously shared `dev-requirements.txt` between conda and pip (so if we included the dependency there, 1 of 2 always failed trying to install `line-profiler` or `line_profiler`). Now that we use the `test` section from the conda recipe, that limitation is no longer an issue.

Reply:
It's apparently still happening.  https://github.com/pcdshub/pcds-ci-helpers/blob/2844a430ae6db9c19d82a213ea8dfa76985d0730/.github/workflows/python-conda-test.yml#L45

Reply:
Fixed via https://github.com/pcdshub/pcds-ci-helpers/pull/99 🎉